modules/gapi/include/opencv2/gapi/util/variant.hpp

```diff
@@ -25,23 +25,27 @@ namespace util
         template<std::size_t I, typename Target, typename First, typename... Remaining>
         struct type_list_index_helper
         {
-            static const constexpr bool is_same = std::is_same<Target, First>::value;
-            static const constexpr std::size_t value =
+            // constexpr removed for C++17 compatibility (SYCL compiler)
+            static const bool is_same = std::is_same<Target, First>::value;
+            // constexpr removed for C++17 compatibility (SYCL compiler)
+            static const std::size_t value =
```

Comment:
Currently, we have been compiling the entire GAPI module with the SYCL compiler (which only supports C++17). It would be great if we could limit this to a smaller subset of the module, but I don't know how that would be accomplished.